### PR TITLE
npcx: move scfg driver from pinctrl to soc/arm/nuvoton_npcx/common

### PR DESCRIPTION
**Pull request by ChiHuaL**
In NPCX chips, System Configuration module can configure not only
pinctrl but also misc. functionality such as glue and flash write
protection. This change moves the scfg driver from the pinctrl folder
to soc/arm/nuvoton_npcx/common and renames it to avoid confusion.

Signed-off-by: Jun Lin <CHLin56@nuvoton.com>

### CODE REVIEW

**Review by keith-zephyr — CHANGES_REQUESTED** [review]
I think you can delete CONFIG_PINMUX=y from the NPCX board files (npcx7m6fb_evb_dfconfig)

**Comment by ChiHuaL** [comment]
> I think you can delete CONFIG_PINMUX=y from the NPCX board files (npcx7m6fb_evb_dfconfig)

Done.

**Comment by galak** [comment]
dev-review: (Mar 11, 2021): moving code makes sense, suggest to remove the 'struct define' / DEVICE_DT_DEFINE and just have this a set of library functions that get called.

File: dts/bindings/pinctrl/nuvoton,npcx-scfg.yaml [hunk 1]
```diff
@@ -3,7 +3,7 @@
 
 description: Nuvoton, NPCX Pin-Controller (Pinmux, 1.8V support and so on) node
```
Comment (mnkp) [inline dts/bindings/pinctrl/nuvoton,npcx-scfg.yaml:4]: We could change the description to state that this is a "System Configuration module"

File: soc/arm/nuvoton_npcx/common/scfg.c [hunk 2]
```diff
@@ -4,7 +4,7 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-#define DT_DRV_COMPAT nuvoton_npcx_pinctrl
+#define DT_DRV_COMPAT nuvoton_npcx_scfg
 
 #include <drivers/pinmux.h>
```
Comment (mnkp) [inline soc/arm/nuvoton_npcx/common/scfg.c:9]: The include should be removed.

File: soc/arm/nuvoton_npcx/common/scfg.c [hunk 3]
```diff
@@ -123,8 +123,8 @@ static int npcx_pinctrl_init(const struct device *dev)
 }
 
 DEVICE_DT_DEFINE(DT_NODELABEL(scfg),
```
Comment (mnkp) [inline soc/arm/nuvoton_npcx/common/scfg.c:125]: Can be replaced by `SYS_INIT`.

**Review by mnkp — CHANGES_REQUESTED** [review]

File: soc/arm/nuvoton_npcx/common/scfg.c (hunk 2 above)
Comment (mnkp) [reply soc/arm/nuvoton_npcx/common/scfg.c:7]: It looks like, the include hasn't been removed yet.

File: dts/bindings/pinctrl/nuvoton,npcx-scfg.yaml (hunk 1 above)
Comment (ChiHuaL) [reply dts/bindings/pinctrl/nuvoton,npcx-scfg.yaml]: Done.

File: soc/arm/nuvoton_npcx/common/scfg.c (hunk 2 above)
Comment (ChiHuaL) [reply soc/arm/nuvoton_npcx/common/scfg.c]: Done.

File: soc/arm/nuvoton_npcx/common/scfg.c (hunk 3 above)
Comment (ChiHuaL) [reply soc/arm/nuvoton_npcx/common/scfg.c]: Done.

**Comment by ChiHuaL** [comment]
> dev-review: (Mar 11, 2021): moving code makes sense, suggest to remove the 'struct define' / DEVICE_DT_DEFINE and just have this a set of library functions that get called.

Done.

**Review by mnkp — APPROVED** [review]

**Review by mbolivar-nordic — APPROVED** [review]

**Review by keith-zephyr — APPROVED** [review]

**Review by jackrosenthal — APPROVED** [review]

**Pull request merged**